src/services/trackingPlan.ts

```diff
@@ -1,88 +1,86 @@
 import logger from '../logger';
 import { RetryRequestError, RespStatusError, constructValidationErrors } from '../util/utils';
-import { getMetadata } from '../v0/util';
+import { getMetadata, getTrackingPlanMetadata } from '../v0/util';
 import eventValidator from '../util/eventValidation';
 import stats from '../util/stats';
+import { HTTP_STATUS_CODES } from '../v0/util/constant';
 
 export class TrackingPlanservice {
-  public static async validateTrackingPlan(events, requestSize, reqParams) {
-    const requestStartTime = new Date();
+  public static async validate(events, requestSize, reqParams) {
+    const startTime = Date.now();
     const respList: any[] = [];
-    const metaTags = events[0].metadata ? getMetadata(events[0].metadata) : {};
+    const metaTags = events.length && events[0].metadata ? getMetadata(events[0].metadata) : {};
+    const tpTags: any =
+      events.length && events[0].metadata ? getTrackingPlanMetadata(events[0].metadata) : {};
     let ctxStatusCode = 200;
 
-    for (let i = 0; i < events.length; i++) {
-      const event = events[i];
-      const eventStartTime = new Date();
+    for (let event of events) {
```

Comment:
Isn't this line causing an ESLint error, suggesting the use of iterators instead of loops?

Reply:
eslint has been blocked for the trackingPlan files. I'll change it nonetheless, point has been raised before as well.